src/routes/InventoryDetail.js

```diff
@@ -98,30 +108,37 @@ const Inventory = () => {
   const store = useStore();
   const navigate = useNavigate();
   const dispatch = useDispatch();
-
+  const [availableApps, setAvailableApps] = useState([]);
   const entityLoaded = useSelector(
     ({ entityDetails }) => entityDetails?.loaded
   );
   const entity = useSelector(({ entityDetails }) => entityDetails?.entity);
   const { cloud_provider: cloudProvider, host_type: hostType } = useSelector(
     ({ systemProfileStore }) => systemProfileStore?.systemProfile || []
   );
-  const availableApps = useMemo(
-    () =>
-      appList.map((app) =>
-        app.name === 'ros'
-          ? {
-              ...app,
-              isVisible: cloudProvider === 'aws',
-            }
-          : app
-      ),
-    [cloudProvider]
-  );
+  useEffect(() => {
+    let osSlug = entity?.system_profile?.operating_system?.name
+      .replace(' ', '-')
+      .toUpperCase();
+
+    let newApps =
+      entity &&
+      appList[osSlug]?.map((app) => {
+        app.isDisabled = app.nonEdge && hostType === 'edge' ? true : false;
+        app['data-cy'] = `${app.name}-tab`;
+        app.name === 'ros' && {
+          ...app,
+          isVisible: cloudProvider === 'aws',
+        };
+        return app;
+      });
+
+    setAvailableApps(newApps);
+  }, [entity, cloudProvider]);
 
   useEffect(() => {
     if (searchParams.get('appName') === null) {
-      setSearchParams('appName', appList[0].name);
+      setSearchParams('appName', appList['RHEL'][0].name);
```

Comment:
This should probably also be something like `appList[osSlug][0].name`.

Reply:
Since this runs once on page load, we need this to be a set value from the jump. We dont want to rerender the page off of the params here. Since General information will most likely almost come first, I kept it as is. 
If we really want to keep away from that, what about setting it to 'general_information'?

Reply:
The `searchParams.get('appName') === null` should prevent this state update to happen, even if `entity` is a dependency of this effect as well to get the right OS name/slug.


TBH. I think this useEffect here is a bit odd. I think it should either update the state, or navigate, but not both. When the navigate is called, it probably throws away this state update here, but it will then have a correct tab selected, because it then has the `appName` set in the URL.


I believe instead of doing any state update or navigate here, we should just default to the first app in the app array of each OS.

Reply:
I dont fully understand what you mean in the last sentence. What exactly are you saying we are defaulting? Are you saying the appName in the url should default to that?

Reply:
@adonispuente What I mean is, if there is no `appName` in the URL initially, when a user comes first to the page, we should not set any `appName` parameter in the URL, but simply default to the first "app" in the array for each OS. We should only set a appName in the URL if a user does click a tab.